File: api-java/src/main/java/com/wpp_message/features/wpp/connect_api/auth/WppConnectAPIAuthService.java
package com.wpp_message.features.wpp.connect_api.auth;

import com.wpp_message.features.wpp.connect_api.auth.requests.WppConnectAPIStartSessionRequest;
import com.wpp_message.features.wpp.connect_api.auth.responses.*;
import io.quarkus.rest.client.reactive.NotBody;
import jakarta.ws.rs.*;
import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;

/**
 * Endpoints para interagir com os métodos de autenticação da API do wppconnect-server
 * https://github.com/wppconnect-team/wppconnect-server
 */
@RegisterRestClient(configKey = "wppconnect-api")
public interface WppConnectAPIAuthService {

    @POST
    @Path(WppConnectAPIAuthServiceHandle.SESSION + "/{secretKey}/generate-token")
    WppConnectAPITokenResponse generateToken(@PathParam("secretKey") String secretKey) throws WebApplicationException;

    @POST
    @Path(WppConnectAPIAuthServiceHandle.SESSION + "/start-session")
    @ClientHeaderParam(name = "Authorization", value = "Bearer {token}")
    WppConnectAPIStartSessionResponse startSession(WppConnectAPIStartSessionRequest request, @NotBody String token) throws WebApplicationException;

    @GET
    @Path(WppConnectAPIAuthServiceHandle.SESSION + "/check-connection-session")
    @ClientHeaderParam(name = "Authorization", value = "Bearer {token}")
    WppConnectAPICheckConnectionSessionResponse checkConnectionSession(@NotBody String token) throws WebApplicationException;

    @POST
    @Path(WppConnectAPIAuthServiceHandle.SESSION + "/close-session")
    @ClientHeaderParam(name = "Authorization", value = "Bearer {token}")
    WppConnectAPICloseSessionResponse closeSession(@NotBody String token) throws WebApplicationException;

    @POST
    @Path(WppConnectAPIAuthServiceHandle.SESSION + "/logout-session")
    @ClientHeaderParam(name = "Authorization", value = "Bearer {token}")
    WppConnectAPILogoutSessionResponse logoutSession(@NotBody String token) throws WebApplicationException;

}

File: web-angular/src/app/api/wpp-connect-api-auth.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { WppConnectAPICloseSessionResponse } from "src/app/core/responses/wpp-connect-api-close-session-response";
import { WppConnectAPILogoutSessionResponse } from "src/app/core/responses/wpp-connect-api-logout-session-response";
import { WppConnectAPIStartSessionResponse } from "src/app/core/responses/wpp-connect-api-start-session-response";
import { WppConnectAPIStatusMessageResponse } from "src/app/core/responses/wpp-connect-api-status-message-response";
import { environment } from "src/environments/environment";

@Injectable()
export class WppConnectAPIAuthService {

  /**
 * @description Armazena a url base do sistema
 */
  public baseUrl = environment.api;

  private _fullUrl: string = '';

  constructor(
    public http: HttpClient,
  ) {
    this._fullUrl = `${this.baseUrl}/wpp-connect-api/auth`;
  }

  public startSession(): Observable<WppConnectAPIStartSessionResponse> {
    return this.http.get<WppConnectAPIStartSessionResponse>(`${this._fullUrl}/start-session`);
  }

  public checkConnectionSession(): Observable<WppConnectAPIStatusMessageResponse> {
    return this.http.get<WppConnectAPIStatusMessageResponse>(`${this._fullUrl}/check-connection-session`);
  }

  public closeSession(): Observable<WppConnectAPICloseSessionResponse> {
    return this.http.get<WppConnectAPICloseSessionResponse>(`${this._fullUrl}/close-session`);
  }

  public logoutSession(): Observable<WppConnectAPILogoutSessionResponse> {
    return this.http.get<WppConnectAPILogoutSessionResponse>(`${this._fullUrl}/logout-session`);
  }

}
